Sources/XMTPiOS/Contacts.swift

```diff
@@ -55,12 +55,13 @@ public class ConsentList {
 		identifier = try? LibXMTP.generatePrivatePreferencesTopicIdentifier(privateKey: privateKey)
 	}
 
-	func load() async throws -> ConsentList {
+	func load(pagination: Pagination? = nil) async throws -> ConsentList {
 		guard let identifier = identifier else {
 			throw ContactError.invalidIdentifier
 		}
 
-		let envelopes = try await client.apiClient.envelopes(topic: Topic.preferenceList(identifier).description, pagination: Pagination(direction: .ascending))
+		let pagination = pagination ?? Pagination(direction: .ascending)
+		let envelopes = try await client.apiClient.envelopes(topic: Topic.preferenceList(identifier).description, pagination: pagination)
```

Comment:
Isn't this already paginated under the hood?

What problem are we solving by putting the user in control of pagination here? When would they want to do anything other than load the entire consent list? Anything less would leave them in an out of date state.

Reply:
Updated to only take an after date to allow for more performant local cache updates

Reply:
I think there needs to be a bigger change here because the refreshConsentList just wipes out the current list and makes a new list. I think the ideal scenario is that the `consentList` doesn't get reset. On every load and that you could just fetch the latest consent items from the last time that you fetched consent. But I think you'll need more changes for that.

Reply:
This is kinda my issue with current consent, I think it's nuanced (at least personally)

Reply:
An integrator would only use afterDate if they were caching the consent list, so they likely would not be using the SDK memory cached consent list after refreshing

Reply:
> An integrator would only use afterDate if they were caching the consent list

Ahh. I see. But our SDK also needs an up-to-date consent list, doesn't it? Is the idea that the integrator would pass their known entries back in to us from their cache on startup?

Alternatively, we could just handle caching for the integrator. I know on iOS there's all kinds of permissions implications, but maybe it's worthwhile anyways.